python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -932,6 +932,29 @@ def _ensure_persisted_execution_plan_snapshot(
 
         return execution_plan_snapshot_id
 
+    def _register_run_asset_events(self, pipeline_run, execution_plan_snapshot):
+        from dagster.core.events import DagsterEvent
+        from dagster.core.events.log import EventLogEntry
+
+        pipeline_name = pipeline_run.pipeline_name
+
+        for step in execution_plan_snapshot.steps:
+            if step.key in execution_plan_snapshot.step_keys_to_execute:
+                for output in step.outputs:
+                    asset_key = output.properties.asset_key
+                    if asset_key:
+                        event = DagsterEvent.register_run_asset(pipeline_name, asset_key)
+                        event_record = EventLogEntry(
+                            user_message="",
+                            level=logging.INFO,
+                            pipeline_name=pipeline_name,
+                            run_id=pipeline_run.run_id,
+                            error_info=None,
+                            timestamp=time.time(),
+                            dagster_event=event,
+                        )
+                        self.handle_new_event(event_record)
```

Comment:
this diverges from the pattern normal to the static constructors on `DagsterEvent` - i assume to avoid logging all the output? 

If we had 1 instead of N events it seems we could avoid going out of our way to hide this information which users may find valuable in some circumstances

Reply:
what if we just dropped he log level to `debug` for "hiding" since the default dagit experience is to hide debug logs (i believe)? This does seem like useful debug information.

Reply:
Yep, I think this will be useful debug information, I'll switch the log level to `debug` instead. The pattern diverges from the static constructors on `DagsterEvent` because no `pipeline_context` or `step_context` object has been created.

Reply:
Hm...seems like we display all `DagsterEvent`s by default in Dagit and only filter debug logs that are not `DagsterEvent`s. I could add special casing though on the front end to hide events of this type